mimic/rest/maas_api.py

```diff
@@ -0,0 +1,960 @@
+"""
+MAAS Mock API
+"""
+
+import json
+import collections
+import time
+import random
+import re
+from uuid import uuid4
+
+from six import text_type
+
+from zope.interface import implementer
+
+from twisted.web.server import Request
+from twisted.plugin import IPlugin
+
+from mimic.catalog import Entry
+from mimic.catalog import Endpoint
+from mimic.rest.mimicapp import MimicApp
+from mimic.imimic import IAPIMock
+from mimic.canned_responses.maas_json_home import json_home
+from mimic.canned_responses.maas_monitoring_zones import monitoring_zones
+from mimic.util.helper import random_hex_generator
+
+
+Request.defaultContentType = 'application/json'
+
+
+@implementer(IAPIMock, IPlugin)
+class MaasApi(object):
+
+    """
+    Rest endpoints for mocked MAAS Api.
+    """
+
+    def __init__(self, regions=["ORD"]):
+        """
+        Set regions
+        """
+        self._regions = regions
+
+    def catalog_entries(self, tenant_id):
+        """
+        List catalog entries for the Nova API.
+        """
+        return [
+            Entry(
+                tenant_id, "rax: monitor", "cloudMonitoring",
+                [
+                    Endpoint(tenant_id, region, text_type(uuid4()),
+                             "v1.0")
+                    for region in self._regions
+                ]
+            )
+        ]
+
+    def resource_for_region(self, region, uri_prefix, session_store):
+        """
+        Get an : obj: `twisted.web.iweb.IResource` for the given URI prefix;
+        implement : obj: `IAPIMock`.
+        """
+        return MaasMock(self, uri_prefix, session_store, region).app.resource()
+
+
+class M_Cache(dict):
```

Comment:
This is a copy of a really unfortunate anti-pattern.  Please file an issue in the vein of #70.